ccx_messaging/consumers/kafka_consumer.py

```diff
@@ -45,11 +43,19 @@ def __init__(
         )
 
         # Confluent initialization
-        config = {
-            "bootstrap.servers": kwargs.get("bootstrap_servers", ""),
-            "group.id": kwargs.get("group_id", None),
-            "retry.backoff.ms": retry_backoff_ms,
-        }
+        
+        config = kwargs.copy()
+        config.update(
+            {
+                "bootstrap.servers": kwargs.get("bootstrap_servers", ""),
+                "group.id": kwargs.get("group_id", None),
+                "retry.backoff.ms": retry_backoff_ms,
+            }
+        )
+        if "bootstrap_servers" in config:
```

Comment:
not sure I understand these deletions. You're adding these values to the dict before, then deleting them if they exist